tests/lgdo/test_lh5_store.py

```diff
@@ -0,0 +1,202 @@
+    assert nda['baseline'].shape == (6,)
+    assert isinstance(nda['waveform/values'], np.ndarray)
+    assert nda['waveform/values'].shape == (6, 5592)
+
+
+def test_load_dfs(lgnd_file):
+    dfs = lh5.load_dfs([lgnd_file, lgnd_file],
+                       ['baseline', 'waveform/t0'],
+                       lh5_group='/geds/raw',
+                       idx_list=[[1, 3, 5], [2, 6, 7]])
+
+    assert isinstance(dfs, pd.DataFrame)
+
+
+@pytest.fixture(scope='module')
+def lh5_file():
+    store = LH5Store()
+
+    struct = lgdo.Struct()
+    struct.add_field('scalar', lgdo.Scalar(value=10))
+    struct.add_field('array', lgdo.Array(nda=np.array([1, 2, 3, 4, 5])))
+    struct.add_field('aoesa', lgdo.ArrayOfEqualSizedArrays(shape=(5, 5), dtype=np.float32, fill_val=42))
+    struct.add_field('vov', lgdo.VectorOfVectors(
+        flattened_data=lgdo.Array(nda=np.array([1, 2, 3, 4, 5, 2, 4, 8, 9, 7, 5, 3, 1])),
+        cumulative_length=lgdo.Array(nda=np.array([2, 5, 6, 10, 13]))
+    ))
+
+    col_dict = {
+        'a': lgdo.Array(nda=np.array([1, 2, 3, 4])),
+        'b': lgdo.Array(nda=np.array([5, 6, 7, 8]))
+    }
+
+    struct.add_field('table', lgdo.Table(col_dict=col_dict))
+
+    store.write_object(struct, 'struct',
+                       '/tmp/tmp-pygama-lgdo-types.lh5',
+                       group='/data',
+                       start_row=1, n_rows=3,
+                       wo_mode='overwrite_file')
+
+    return '/tmp/tmp-pygama-lgdo-types.lh5'
+
+
+def test_write_objects(lh5_file):
+    pass
+
+
+def test_read_scalar(lh5_file):
+    store = LH5Store()
+    lh5_obj, n_rows = store.read_object('/data/struct/scalar', lh5_file)
+    assert isinstance(lh5_obj, lgdo.Scalar)
+    assert lh5_obj.value == 10
+    assert n_rows == 1
+
+
+def test_read_array(lh5_file):
+    store = LH5Store()
+    lh5_obj, n_rows = store.read_object('/data/struct/array', lh5_file)
+    assert isinstance(lh5_obj, lgdo.Array)
+    assert (lh5_obj.nda == np.array([2, 3, 4])).all()
+    assert n_rows == 3
+
+
+def test_read_vov(lh5_file):
+    store = LH5Store()
+    lh5_obj, n_rows = store.read_object('/data/struct/vov', lh5_file)
+    assert isinstance(lh5_obj, lgdo.VectorOfVectors)
+
+    desired = [np.array([3, 4, 5, 2]),
+               np.array([]),
+               np.array([])]
+
+    for i in range(len(desired)):
+        assert (desired[i] == list(lh5_obj)[i]).all()
+
+    assert n_rows == 3
+
+
+def test_read_aoesa(lh5_file):
+    store = LH5Store()
+    lh5_obj, n_rows = store.read_object('/data/struct/aoesa', lh5_file)
+    assert isinstance(lh5_obj, lgdo.ArrayOfEqualSizedArrays)
+    assert (lh5_obj.nda == np.full((3, 5), fill_value=42)).all()
+
+
+def test_read_table(lh5_file):
+    store = LH5Store()
+    lh5_obj, n_rows = store.read_object('/data/struct/table', lh5_file)
+    assert isinstance(lh5_obj, lgdo.Table)
+    assert n_rows == 3
+
+    # TODO: test failure here?
+    # store.read_object('/data/struct/table', [lh5_file, lh5_file])
```

Comment:
@jasondet wouldn't it be maybe nice to raise an exception if one tries to concatenate non-array stuff? I'm not sure how to implement this in `read_object` though...

Reply:
What's the specific error it's giving? What is in the table that is non-array?

Reply:
This is what I get from `store.read_object('/geds/raw/waveform', [lh5_file, lh5_file])`, where `lh5_file` is `legend-testdata/data/lh5/LDQTA_r117_20200110T105115Z_cal_geds_raw.lh5`
```
---------------------------------------------------------------------------
MemoryError                               Traceback (most recent call last)
Input In [6], in <cell line: 1>()
----> 1 store.read_object('/geds/raw/waveform', [lh5_file, lh5_file])

File ~/.virtualenvs/pygama-vccw/lib/python3.10/site-packages/pygama/lgdo/lh5_store.py:204, in LH5Store.read_object(self, name, lh5_file, start_row, n_rows, idx, field_mask, obj_buf, obj_buf_start, verbosity)
    202 else: idx_i = None
    203 n_rows_i = n_rows-n_rows_read
--> 204 obj_buf, n_rows_read_i = self.read_object(name,
    205                                           lh5_file[i],
    206                                           start_row=start_row,
    207                                           n_rows=n_rows_i,
    208                                           idx=idx_i,
    209                                           field_mask=field_mask,
    210                                           obj_buf=obj_buf,
    211                                           obj_buf_start=obj_buf_start,
    212                                           verbosity=verbosity)
    213 n_rows_read += n_rows_read_i
    214 if n_rows_read >= n_rows or obj_buf == None:

File ~/.virtualenvs/pygama-vccw/lib/python3.10/site-packages/pygama/lgdo/lh5_store.py:344, in LH5Store.read_object(self, name, lh5_file, start_row, n_rows, idx, field_mask, obj_buf, obj_buf_start, verbosity)
    335     col_dict[field], n_rows_read = self.read_object(name+'/'+field,
    336                                                     h5f,
    337                                                     start_row=start_row,
   (...)
    341                                                     obj_buf_start=obj_buf_start,
    342                                                     verbosity=verbosity)
    343     if obj_buf is not None and obj_buf_start+n_rows > len(obj_buf):
--> 344         obj_buf.resize(obj_buf_start+n_rows, do_warn=(verbosity>0))
    345     rows_read.append(n_rows_read)
    346 # warn if all columns don't read in the same number of rows

File ~/.virtualenvs/pygama-vccw/lib/python3.10/site-packages/pygama/lgdo/table.py:80, in Table.resize(self, new_size, do_warn)
     77             print('warning: resizing field', field,
     78                   'with size', len(obj), '!=', new_size)
     79         if isinstance(obj, Table): obj.resize(new_size, do_warn)
---> 80         else: obj.resize(new_size)
     81 self.size = new_size

File ~/.virtualenvs/pygama-vccw/lib/python3.10/site-packages/pygama/lgdo/array.py:82, in Array.resize(self, new_size)
     80 """Resize the array to new_size (int)"""
     81 new_shape = (new_size,) + self.nda.shape[1:]
---> 82 self.nda.resize(new_shape, refcheck=True)

MemoryError:
```